bigtop-packages/src/common/ignite-hadoop/do-component-build

```diff
@@ -21,4 +21,30 @@ set -ex
 export MAVEN_OPTS="-Xmx512M"
 
 sed -i '/asm-all/{n;s/4.2/5.0.2/}' modules/hadoop/pom.xml
+
+if [ "$HOSTTYPE" != "x86_64" ] ; then
+    # download ignite-shmem from maven central
+    mvn dependency:get -DremoteRepositories=https://repo1.maven.org/maven2/ -DgroupId=org.gridgain -DartifactId=ignite-shmem -Dversion=1.0.0
+    mkdir tmp-build
+    cd tmp-build
+    # download source jar
+    wget https://repo1.maven.org/maven2/org/gridgain/ignite-shmem/1.0.0/ignite-shmem-1.0.0-sources.jar
```

Comment:
Any chance we can evaluate the signature for this file purely obtained from internet?
It should be fairly easy since the version of jar is already fixed, the signature to check can be fixed here as well.

Reply:
Right beside the file there is an .asc file. Evans, you may proposed the same approach as I did for ant in puppet code ?

"/usr/bin/gpg -v --verify --auto-key-retrieve --keyserver hkp://keyserver.ubuntu.com:80 $ant-bin.tar.gz.asc"

This would be rather lame, since we would test sources by comparing with a file retrieved from the same location. Situation is different for the ant scenario since we download from an untrusted party.

Reply:
@oflebbe yup downloading from the same remote seems no much help to security. I think having the .asc file content hard-coded in our code base should be OK.

Reply:
@evans-ye @oflebbe Would it be OK to verify both sha1 and md5 checksum of the jar file, and compare them with hardcoded in our code base?

Reply:
That's what I'm proposing. It should be fairly easy to implement. If possible, let's try this and get this in.